### PR TITLE
(MODULES-5433) update docs to include iis_virtual_directory

### PR DESCRIPTION
**Pull request by tkishel**
Correct Types and Providers anchor
Add iis_virtual_directory section
Replace DefaultApplicationPool with DefaultAppPool
Normalize case of started and stopped values
Normalize value formatting: quotes versus backticks

### CODE REVIEW

File: README.md
Comment (jbondpdx) [inline README.md]: present and absent are correct in single quotes, as they are strings rather than values like `true` or `false`.

File: README.md
Comment (jbondpdx) [inline README.md]: These options are strings, so should be in single quotes.

File: README.md
Comment (jbondpdx) [inline README.md]: these options are strings, so single quotes are correct

File: README.md
Comment (jbondpdx) [inline README.md]: These are string values, so single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: these values are strings, so should be single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: Valid options for this are both passed as strings, so should be in single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: options are string values, should be single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: strings, single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: value SpecificUser is passed as a string, so single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: strings, single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: strings, single quotes

File: README.md
Comment (jbondpdx) [inline README.md]: strings, single quotes

**Review by jbondpdx — CHANGES_REQUESTED** [review]
Great changes, thank you! Only thing is that a lot of values are passed as strings, so they should be in single quotes rather than backticks.

**Comment by tkishel** [comment]
Thanks so much for the clarification and that level of detail. You could have just clarified the rule and I would have applied to all the instances :) I've reviewed the entire document and found other instances, and other typos and arbitrary differences. I'll rebase and push. If this exceed the scope of the pull request and/or ticket, please let me know that and how to proceed. Thanks again!

**Review by jbondpdx — APPROVED** [review]

**Comment by jbondpdx** [comment]
Yeah, @tkishel , I didn't mean to beat it to death, but I also hate to make the person hunt for all of them. Though apparently, I didn't catch them all anyway. :-) Anyway, these changes look great to me, thank you!

File: README.md
Comment (glennsarti) [inline README.md:75]: This comment should stay in as it is not obvious where the `acl` resource is coming from.

**Comment by tkishel** [comment]
I didn't notice any beating, and yay!

**Comment by jbondpdx** [comment]
I hit up @glennsarti to check the code samples, and we can merge after that. Thanks again @tkishel !

File: README.md
Comment (glennsarti) [inline README.md]: `->` not required due to autorequires

File: README.md
Comment (glennsarti) [inline README.md]: `->` not required due to autorequires

File: README.md
Comment (glennsarti) [inline README.md]: Should probably move `iis_feature` back as this breaks the alphabetical ordering.

File: README.md
Comment (glennsarti) [inline README.md]: `iis_feature` should be after `iis_application_pool`

File: README.md
Comment (glennsarti) [inline README.md:165]: @jbondpdx I'm thinking that `application` should be capitalized.  Reasoning is, that is the name of thing we are managing.  This would probably also explain the use of `IIS Web Application` below (although, the capitalization of that is inconsistent)

File: README.md
Comment (glennsarti) [inline README.md]: The use of the word `including` here could be ambiguous.  The `sslflags` can have one or more of those 4 options. Possible rewording:

`Valid options are an array of flags, with the following names 'Ssl'....`

File: README.md
Comment (glennsarti) [inline README.md]: See comments above

**Review by glennsarti — CHANGES_REQUESTED** [review]
Also the readme text here is normally derived from the inline help in the various type defintions e.g. https://github.com/puppetlabs/puppetlabs-iis/blob/master/lib/puppet/type/iis_application.rb

Normally we would update both so they are consistent.

File: README.md
Comment (tkishel) [reply README.md:75]: I didn't delete that comment, I moved it out of the code and into the "This complete example ..." paragraph above @ Line 76. Is that okay, or should I move it back?

File: README.md
Comment (tkishel) [reply README.md]: Done

File: README.md
Comment (tkishel) [reply README.md]: Done

File: README.md
Comment (tkishel) [reply README.md]: Done. 

I was unaware of our preference for alphabetical ordering, and the alphabetical ordering is almost identical to the relationship ordering (you need iis_feature before iis_*)

File: README.md
Comment (tkishel) [reply README.md]: Done

File: README.md
Comment (jbondpdx) [reply README.md:165]: Microsoft and their darned caps, man. "application" feels like it could go either way, depending on whether we are using as the generic noun or the MS-specific proper name Application. In this case, I'm inclined to defer to your Microsoft expertise on this one.

File: README.md
Comment (tkishel) [reply README.md]: Done

File: README.md
Comment (tkishel) [reply README.md]: Done

File: README.md
Comment (tkishel) [reply README.md:165]: This (referring to the object) is something that was inconsistent throughout this document. It used "an | the" with "object | Object | IIS object | IIS Object | IIS Web Object". If you provide a (different) standard, I will normalize.

File: README.md
Comment (tkishel) [reply README.md:75]: @glennsarti ^^

File: README.md
Comment (tkishel) [reply README.md:165]: @glennsarti @jbondpdx  ^^

**Comment by tkishel** [comment]
Thanks Glenn!

File: README.md
Comment (glennsarti) [reply README.md:75]: Ahh ok.  Missed that.

File: README.md
Comment (glennsarti) [reply README.md:165]: Happy to leave it as `application` for this PR.  At least this way it's consistent.

Comment (tkishel) [reply README.md:165]: At least it can now be search and replaced, in the future.

**Comment by tkishel** [comment]
RE: Also the readme text here is normally derived from the inline help ... normally we would update both so they are consistent. I referred to the type definitions while updating this, and tried to align with them.

I don't know how that autogeneration works, but I'm willing to reconcile them. There are a flurry of PRs to the code (a few of them are mine) so I would rather wait until whichever ones will be accepted are merged first. Having been able to review all of the text consolidated and normalized in the README, I expect that flowing changes back into the types will result in more consistent type definition descriptions.

**Comment by glennsarti** [comment]
Yeah, unfortunately there is no automation between the readme and the help text in the types.  All manual.

**Comment by glennsarti** [comment]
@tkishel Changes look good.  if you can squash, I can merge

**Review by glennsarti — APPROVED** [review]

**Pull request merged**
